java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -86,73 +86,88 @@ private FlowComputation(@Nullable SymbolicValue symbolicValue, Predicate<Constra
     return flowComputation.run(currentNode, trackSymbol);
   }
 
-  private static class NodeSymbol {
-    final ExplodedGraph.Node node;
-    final Symbol trackSymbol;
+  private static class Path {
```

Comment:
I addressed the name, also moved most of the flow computation logic into this class and made it immutable making code bit cleaner. The rest is hard to address, I would argue that it's not a new concept, just making what was there before bit more explicit.